packages/roosterjs-editor-core/lib/coreApi/selectTable.ts

```diff
@@ -0,0 +1,202 @@
+import { getTagOfNode, toArray } from 'roosterjs-editor-dom';
+import {
+    EditorCore,
+    SelectionRangeTypes,
+    TableSelection,
+    SelectTable,
+    Coordinates,
+} from 'roosterjs-editor-types';
+
+const TABLE_ID = 'tableSelected';
+const CONTENT_DIV_ID = 'contentDiv_';
+
+let styleElement: HTMLStyleElement;
+
+/**
+ * @internal
+ * Select a table and save data of the selected range
+ * @param core The EditorCore object
+ * @param table table to select
+ * @param coordinates first and last cell of the selection, if this parameter is null, instead of
+ * selecting, will unselect the table.
+ * @returns true if successful
+ */
+export const selectTable: SelectTable = (
+    core: EditorCore,
+    table: HTMLTableElement,
+    coordinates?: TableSelection
+) => {
+    unselect();
+
+    if (coordinates && table) {
+        ensureUniqueId(table, TABLE_ID);
+        ensureUniqueId(core.contentDiv, CONTENT_DIV_ID);
+
+        const ranges = select(core, table, coordinates);
+        return {
+            type: SelectionRangeTypes.TableSelection,
+            ranges,
+            table,
+            areAllCollapsed: ranges.filter(range => range?.collapsed).length == ranges.length,
+            coordinates,
+        };
+    }
+
+    return null;
+};
+
+function buildCss(
+    table: HTMLTableElement,
+    coordinates: TableSelection,
+    contentDivSelector: string
+): { css: string; ranges: Range[] } {
+    coordinates = normalizeTableSelection(coordinates, table);
+    const tr1 = coordinates.firstCell.y + 1;
+    const td1 = coordinates.firstCell.x + 1;
+    const tr2 = coordinates.lastCell.y + 1;
+    const td2 = coordinates.lastCell.x + 1;
+    const ranges: Range[] = [];
+
+    let firstSelected: HTMLTableCellElement | null = null;
+    let lastSelected: HTMLTableCellElement | null = null;
+    let css = '';
+    let isFirst = true;
+
+    // Get whether table has thead, tbody or tfoot.
+    const tableChilds = toArray(table.childNodes).filter(
+        node => ['THEAD', 'TBODY', 'TFOOT'].indexOf(getTagOfNode(node)) > -1
+    );
+
+    // Set the start and end of each of the table childs, so we can build the selector according the element between the table and the row.
+    let cont = 0;
+    const indexes = tableChilds.map(node => {
```

Comment:
The calculation of indexes can be an issue when table has merged cells. We need VTable to help get correct TD/TH here. And I think the easiest way to handle multiple THEAD/TBODY/TFOOT/TH is to just make a full loop when we generate the CSS